src/scroll-timeline-css-parser.js

```diff
@@ -123,7 +126,23 @@ export class StyleParser {
       return null;
   }
 
-  getScrollTimelineOptions(timelineName) {
+  getAnonymousScrollTimelineOptions(timelineName, target) {
+    const options = this.anonymousScrollTimelineOptions.get(timelineName);
+    if(options) {
+      return {
+        source: (options.source == 'root' ? document.scrollingElement : getScrollParent(target)),
```

Comment:
By resolving immediately, we won't update the source in response to DOM changes. We have WPT tests for this already for the JS API.  How about making source Element | string/enum and then resolving a non-element source at the same time we validate/update the view timeline source.